tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/LiferayWorkspaceSDKListener.java

```diff
@@ -0,0 +1,123 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+
+package com.liferay.ide.project.core;
+
+import com.liferay.ide.core.util.CoreUtil;
+import com.liferay.ide.project.core.util.ProjectUtil;
+import com.liferay.ide.sdk.core.SDKUtil;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.IResourceChangeEvent;
+import org.eclipse.core.resources.IResourceChangeListener;
+import org.eclipse.core.resources.IResourceDelta;
+import org.eclipse.core.resources.IResourceDeltaVisitor;
+import org.eclipse.core.resources.WorkspaceJob;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+
+/**
+ * @author Simon Jiang
+ */
+public class LiferayWorkspaceSDKListener implements IResourceChangeListener, IResourceDeltaVisitor
+{
+
+    private static final String ID_WORKSPACE_SDK_INVALID = "workspace-sdk-invalid";
+
+    @Override
+    public void resourceChanged( IResourceChangeEvent event )
+    {
+        if( event == null )
+        {
+            return;
+        }
+
+        try
+        {
+            event.getDelta().accept( this );
+        }
+        catch( CoreException e )
+        {
+        }
+    }
+
+    @Override
+    public boolean visit( final IResourceDelta delta ) throws CoreException
+    {
+        switch( delta.getResource().getType() )
+        {
+            case IResource.ROOT:
+            case IResource.FOLDER:
+            case IResource.PROJECT:
+                return true;
+
+            case IResource.FILE:
+            {
+                if ( delta.getKind() == IResourceDelta.ADDED || delta.getKind() == IResourceDelta.REMOVED )
+                {
+                    IFile deltaFile = (IFile)delta.getResource();
+
+                    if ( deltaFile.getName().equals( "build.properties" ))
+                    {
+                        Job job = new WorkspaceJob( "Checking workspace SDK" )
+                        {
+                            @Override
+                            public IStatus runInWorkspace( IProgressMonitor monitor ) throws CoreException
+                            {
+                                boolean findSDK = false;
+                                IProject[] projects = CoreUtil.getAllProjects();
+
+                                for( IProject existProject : projects )
+                                {
+                                    if ( existProject != null )
```

Comment:
get all projects should not return a null project in the array so we shouldn't need this check here.